lib/wasi/src/lib.rs

```diff
@@ -133,138 +231,136 @@ impl WasiEnv {
 
     /// Get a reference to the memory
     pub fn memory(&self) -> &Memory {
-        self.memory_ref()
+        self.memory
+            .get_ref()
             .expect("Memory should be set on `WasiEnv` first")
     }
 
-    pub(crate) fn get_memory_and_wasi_state(
-        &self,
-        _mem_index: u32,
-    ) -> (&Memory, MutexGuard<WasiState>) {
-        let memory = self.memory();
-        let state = self.state.lock().unwrap();
-        (memory, state)
+    // Copy the lazy reference so that when its initialized during the
+    // export phase that all the other references get a copy of it
```

Comment:
```suggestion
    /// Copy the lazy reference so that when it's initialized during the
    /// export phase, all the other references get a copy of it
```